fe/lib/components/LocationSuggest/LocationSuggestInput.tsx

```diff
@@ -82,17 +99,48 @@ const LocationSuggestInput = ({
     }
   };
 
+  const handleDetectLocationClicked = () => {
+    if (navigator.geolocation) {
+      setLocationSuggest(initialLocationSuggest);
+      navigator.geolocation.getCurrentPosition((position) =>
+        onDetectLocation({
+          latitude: position.coords.latitude,
+          longitude: position.coords.longitude,
+        }),
+      );
+    } else {
+      onDetectLocation(
+        new Error('Geolocation is not supported by this browser.'),
+      );
+    }
+  };
+
   return (
-    <Autosuggest
-      icon={<IconSearch />}
-      id="locationSuggestInput"
-      automaticSelection={false}
-      onClear={handleClear}
-      onChange={handleChange}
-      suggestions={highlightedSuggestions}
-      value={locationSuggest}
-      {...restProps}
-    />
+    <Columns space="small">
+      <Column>
+        <Autosuggest
+          placeholder={placeholder}
+          icon={<IconSearch />}
+          id="locationSuggestInput"
+          automaticSelection={false}
+          onClear={handleClear}
+          onChange={handleChange}
+          suggestions={highlightedSuggestions}
+          value={locationSuggest}
+          {...restProps}
+        />
+      </Column>
+      <Column width="content">
+        <Button
+          loading={isLoading}
+          weight="weak"
+          onClick={handleDetectLocationClicked}
+        >
+          Detect location
+          <IconLocation />
```

Comment:
Something a bit picky, but we should add some padding between "Detect location" and the icon. You can wrap either in a Braid box and add padding left/right to it.

Reply:
Is JSX eating the space in between because they aren't on the same line?

Reply:
Yeah, another way to do this is @ak-08 is to add `{' '}` after "Detect location" to force it to render a space instead.

Reply:
`Detect location <IconLocation />`

Doing this does the job 🥳